File: android/src/main/kotlin/com/royzed/simple_bg_location/errors/ErrorCodes.kt
package com.royzed.simple_bg_location.errors

enum class ErrorCodes(val code: String, val description: String) {
    permissionDefinitionsNotFound(
        "PERMISSION_DEFINITIONS_NOT_FOUND",
        "No location permissions are defined in the manifest. "+
                "Make sure at least ACCESS_FINE_LOCATION or ACCESS_COARSE_LOCATION "+
                "are defined in the manifest."
    ),
    permissionDenied(
        "PERMISSION_DENIED",
        "User denied permissions to access the device's location."
    ),
    otherRequestInProgress(
        "OTHER_REQUEST_INPROGRESS",
        "previous request is still waiting answer."
    ),
    activityMissing(
        "ACTIVITY_MISSING",
        "Activity is missing. This might happen when running a certain function from " +
                "the background that requires a UI element (e.g. requesting permissions or "+
                "enabling the location services)."
    ),
    locationServicesDisabled(
        "LOCATION_SERVICES_DISABLED",
        "Neither GPS nor Network location service is enabled."

    )

}

File: ios/Classes/domain/ErrorCodes.swift
//
//  ErrorCodes.swift
//  simple_bg_location
//
//  Created by Roy Wang on 2022-06-23.
//

import Foundation

struct ErrorCode {
    let code: String
    let description: String
}

struct ErrorCodes {
    static let permissionDenied =
    ErrorCode(
        code: "PERMISSION_DENIED",
        description: "User denied permissions to access the device's location.")
    static let locationServicesDisabled =
    ErrorCode(
        code: "LOCATION_SERVICES_DISABLED",
        description: "Neither GPS nor Network location service is enabled.")
    static let errorWhileAcquiringPosition =
    ErrorCode(
        code: "ERROR_WHILE_ACQUIRING_POSITION",
        description: "An unexpected error occurred while trying to acquire the device's position.")
    static let missedCallReadyMethod =
    ErrorCode(
        code: "MISSED_CALL_READY",
        description: "Missed call ready method before call requestPositionUpdate"
    )
}
